### PR TITLE
Updated endpoints, schemas and started seed script

### PR DESCRIPTION
**Pull request by simei-yeh**

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -20,8 +20,8 @@ function App() {
   const [checkingDatesSet, setCheckingDatesSet] = useState(false);
   const [focusedDate, setFocusedDate] = useState(undefined);
   function getListingData() {
-    const listingID = window.location.pathname.split('/')[1];
-    return axios.get(`/api/listings/${isNaN(listingID) ? 0 : listingID}`)
+    const listingID = window.location.pathname.split('/')[2];
+    return axios.get(`/api/listings/${listingID}`)
```
Comment (gregoryrichens) [inline client/src/components/App.jsx:24]: are you planning on only using a url to the proxy server with an attached id?

I just inserted a couple of lines here to handle a default case if you want to use!

`actualID = Number(listingID) || 21
  return axios.get(`api/listings/${actualID}`)`

File: controllers/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    });
+}
+
+
+//must modify
+function addBooking(req, res) {
+  helpers.listingModel.findOneAndUpdate({ id: req.params.id },
+    {$push: { reserved: req.body } }, { new: true} )
+    .then((listings) => {
+      res.header('Content-Type', 'application/json');
```
Comment (gregoryrichens) [inline controllers/index.js:17]: I don't understand setting headers well. TEACH ME!

File: controllers/index.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      res.send(JSON.stringify(listings, 0, 2));
+    });
+}
+
+function modifyBooking(req, res) {
+  helpers.listingModel.find({ id: req.params.id })
+    .then((listings) => {
+      res.header('Content-Type', 'application/json');
+      res.send(JSON.stringify(listings, 0, 2));
+    });
```
Comment (gregoryrichens) [inline controllers/index.js:27]: would suggest adding .catch logic for any errors

File: server/app.js [hunk 4]
```diff
@@ -3,26 +3,17 @@ const express = require('express');
...
-    });
-});
-app.get('/api/listings/:id', (req, res) => {
-  Helpers.listingModel.find({ id: req.params.id })
-    .then((listings) => {
-      res.header('Content-Type', 'application/json');
-      res.send(JSON.stringify(listings, 0, 2));
-    });
-});
+app.get('/api/listings/:id', controllers.getSpecificListing);
```
Comment (gregoryrichens) [inline server/app.js:14]: love the refactor, much cleaner

**Review by gregoryrichens — APPROVED** [review]
Very clean. A couple of suggestions. No errors from my initial pass.

File: client/src/components/App.jsx (hunk 1 above)
Comment (simei-yeh) [reply client/src/components/App.jsx:24]: updated this for next commit

**Pull request merged**